client_plugin/drivers/shared/kvstore/kvstore.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvstore
+
+// VolStatus: Datatype for keeping status of a shared volume
+type VolStatus string
+
+/*
+   Constants:
+   VolStateCreating:     Shared volume is being created. Not ready to be mounted.
+   VolStateReady:        Shared volume is ready to be mounted but
+                         no Samba service running right now.
+   VolStateMounted:      Samba service already running. Volume mounted
+                         on at least one host VM.
+   VolStateIntermediate: Metadata of shared volume is being changed.
+                         Dont proceed with your op if this is the status.
+   VolStateError:        Shared volume in error state.
+
+   VolPrefixState:       Each volume has three metadata keys. Each such
+                         key terminates in the name of the volume, but
+                         is preceded by a prefix. This is the prefix
+                         for "State" key
+   VolPrefixGRef:        The prefix for GRef key (Global refcount)
+   VolPrefixInfo:        The prefix for info key. This key holds all
+                         other metadata fields squashed into one
+
+   VolumeDoesNotExistError:    Error indicating that there is no such volume
+*/
+const (
+	VolStateCreating        VolStatus = "Creating"
+	VolStateReady           VolStatus = "Ready"
+	VolStateMounted         VolStatus = "Mounted"
+	VolStateIntermediate    VolStatus = "MetadataUpdateInProgress"
+	VolStateError           VolStatus = "Error"
+	VolPrefixState                    = "SVOLS_stat_"
+	VolPrefixGRef                     = "SVOLS_gref_"
+	VolPrefixInfo                     = "SVOLS_info_"
+	VolumeDoesNotExistError           = "No such volume"
+)
+
+// KvPair : Key Value pair holder
+type KvPair struct {
+	Key   string
+	Value string
+}
+
+// KvStore is the interface for VolumeDriver to access a plugin-level KV store
+type KvStore interface {
+	// WriteVolMetadata - Update or Create volume metadata in KV store
+	WriteVolMetadata(entries []KvPair) error
+
+	// ReadVolMetadata - Read volume metadata in KV store
+	ReadVolMetadata(keys []string) ([]KvPair, error)
+
+	// DeleteVolMetadata - Delete volume metadata in KV store
+	DeleteVolMetadata(name string) error
```

Comment:
I against remove Vol for the same reason above :)
This is for docker related operations, but not volume related operations.